Orange/widgets/data/oweditdomain.py

```diff
@@ -9,29 +9,35 @@
 from xml.sax.saxutils import escape
 from itertools import zip_longest
 from contextlib import contextmanager
-from collections import namedtuple
+from collections import namedtuple, Counter
 from functools import singledispatch
 
 from typing import (
     Tuple, List, Any, Optional, Union, Dict, Sequence, Iterable, NamedTuple,
+    FrozenSet,
 )
 
 from AnyQt.QtWidgets import (
     QWidget, QListView, QTreeView, QVBoxLayout, QHBoxLayout, QFormLayout,
     QToolButton, QLineEdit, QAction, QActionGroup, QStackedWidget, QGroupBox,
     QStyledItemDelegate, QStyleOptionViewItem, QStyle, QSizePolicy, QToolTip,
-    QDialogButtonBox, QPushButton, QCheckBox
+    QDialogButtonBox, QPushButton, QCheckBox,
+    QComboBox
 )
 from AnyQt.QtGui import QStandardItemModel, QStandardItem, QKeySequence, QIcon
+from AnyQt.QtCore import (
+    Qt, QEvent, QSize, QModelIndex, QAbstractItemModel, QPersistentModelIndex,
+    QRect
+)
 from AnyQt.QtCore import pyqtSignal as Signal, pyqtSlot as Slot
-from AnyQt.QtCore import Qt, QEvent, QSize, QModelIndex
 
 import numpy as np
 
 import Orange.data
 
 from Orange.preprocess.transformation import Identity, Lookup
 from Orange.widgets import widget, gui, settings
+from Orange.widgets.data.owconcatenate import unique
```

Comment:
I'd avoid importing functions from widget modules if possible because it messes up dependencies and complicates future refactorings. I think we'd better treat all functions in widget modules as private (even if not named as such), just as we treat widget classes as final.

Do we have a better place for functions like `unique`?

Perhaps better, isn't `unique(s)` same as `iter(dict.fromkeys(s))` (formally from Python 3.7, de facto already in 3.6). In below code, it's used as `list(unique(s))`, which is, I suppose, the same as `list(dict.fromkeys(s))`.